### PR TITLE
azure - improve getting and naming loggers

### PR DESCRIPTION
**Pull request by erwelch**
This PR:

1. Moves all getLogger() calls out of __init__ for actions and filters
2. Renames class level loggers to have their Class Name
3. Adds class level loggers to classes that are calling `self.log` so we get better debugging experience
4. Cleans up modules that had a mixture of module/class level logging

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/output.py [hunk 1]
```diff
@@ -52,11 +52,12 @@ class AzureStorageOutput(DirectoryOutput):
 
     DEFAULT_BLOB_FOLDER_PREFIX = '{policy_name}/{now:%Y/%m/%d/%H/}'
 
+    log = logging.getLogger('custodian.azure.output')
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/output.py]: Shouldn't this be `AzureStorageOutput`?

File: tools/c7n_azure/c7n_azure/function_package.py [hunk 2]
```diff
@@ -48,9 +48,9 @@ def create_zinfo(self, file):
 
 
 class FunctionPackage(object):
+    log = logging.getLogger('custodian.azure.function_package')
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/function_package.py]: Should this be `FunctionPackage`?

File: tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py [hunk 3]
```diff
@@ -9,12 +9,12 @@
 
 @six.add_metaclass(ABCMeta)
 class DeploymentUnit:
+    log = logging.getLogger('custodian.azure.deployment_unit')
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py]: `DeploymentUnit`?

File: tools/c7n_azure/c7n_azure/resources/storage.py [hunk 4]
```diff
@@ -140,9 +140,10 @@ class StorageSetFirewallAction(SetFirewallAction):
         }
     )
 
+    log = logging.getLogger('custodian.azure..StorageSetFirewallAction')
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/resources/storage.py]: You have a double period here.

File: tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py (hunk 3 above)
Comment (erwelch) [reply tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py]: Yeah see this is where I internally debated module vs class. The file name is `deployment_unit` and there are no other classes in here. For consistency sake if I keep it in the class I'll ad the class name too.

File: tools/c7n_azure/c7n_azure/function_package.py (hunk 2 above)
Comment (erwelch) [reply tools/c7n_azure/c7n_azure/function_package.py]: I only specified the class in files where other major classes were in the module and logging as well.  But for consistency I'll go through and just add the class name to everything.

**Comment by stefangordon** [comment]
Big improvement.
Closes #4644

**Pull request merged**